### PR TITLE
Completely rewrite `textbook` module and its unit tests

### PR DESCRIPTION
**Pull request by tianyizheng02**
Contributes to #45 by fixing all `mypy` errors in `textbook.py` as part of the module rewrite

As discussed in the CSC Discord server, `textbook.py` has been found to be broken. Parts of the underlying textbook API had changed, invalidating most of the existing code. For instance, one major change to the underlying API is that the API's request URLs now require CSRF tokens, which I believe weren't required before. Major breaking changes like this weren't caught earlier because `textbook_test.py` mostly didn't validate any function outputs, only their return types, and thus the module's unit tests merely gave the illusion of correctness.

I rewrote the module along with its unit tests to adjust for the changed textbook API, while taking the liberty to make the code more readable and robust based on discussions in the CSC Discord server. This includes better error handling, input validation, automatic retrieval of CSRF tokens, and much more comprehensive unit tests. The rewritten `textbook.py` passes `mypy --strict --ignore-missing-imports` with no errors and has 90+% code coverage with `pytest`.

In addition to the code changes, I'm also introducing several more sample/mock files in order to properly test the API requests. As part of the changes to the mocks, I replaced the data for the STAT department with data for the MATH department instead—for some reason STAT classes just don't seem to have their textbooks listed on the Pitt textbook site, meaning that there's virtually no textbook data available for those classes that we can use as mocks anyway.

### CODE REVIEW

**Comment by tianyizheng02** [comment]
Future TODO: figure out how the Pitt textbook website computes the term ID for a given academic term. I have the ID for fall 2024 hard-coded into `textbook.py` and this is the best I can do for now, but we should definitely replace this with a function that computes this ID for any academic term. Otherwise, we'll need to change this hard-coded ID whenever a new semester draws near.

**Comment by tianyizheng02** [comment]
@akrishan11 FYI, we should hold off on merging this PR until #197 is merged. This is because #197 fixes an issue with the `grequests` library, and `textbook.py` uses this exact library.

**Comment by tianyizheng02** [comment]
@akrishan11 #197 has now been merged, and `textbook` now imports correctly without any `MonkeyPatchWarning`s or `RecursionError`s. However, that warning/error can still occur if we import any `requests`-related packages before importing `textbook`, but we can't really avoid this.

**Review by akrishan11 — APPROVED** [review]

**Pull request merged**
